LMSBorrower/src/main/java/com/lms/LMSBorrower/BorrowerDAO/BorrowerDAO.java

```diff
@@ -1,16 +1,162 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Calendar;
 
+import org.springframework.stereotype.Component;
+import com.lms.LMSBorrower.POJO.*;
+
+@Component
 public class BorrowerDAO {
 	
 	
+	public Connection getConnection() throws SQLException{
+		  Connection con = null;
+		 try {
+			 con = (Connection) DriverManager.getConnection("jdbc:mysql://localhost:3306/library?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC","root","TapHouse123");
+	         } 
+		 catch(SQLException e) {
+	            System.out.println(e);
+	       }   
+		  return con;
+	    }
 	
-	public Borrower getBorrowerByCardNo(int cardNo) {
+	//get cardNo
+	public Borrower getBorrowerByCardNo(int cardNo) throws SQLException {
+		String sql = "SELECT * FROM tbl_borrower WHERE cardNo =?";
+		PreparedStatement ps = getConnection().prepareStatement(sql);
+		ps.setInt(1,  cardNo);
+		ResultSet rs = ps.executeQuery();
 		Borrower bor = new Borrower();
-		 bor.setCardNo(1);
-		 bor.setName("Sayana");
-		 return bor;
+		
+		while (rs.next()) {
+			bor.setCardNo(rs.getInt("cardNo"));
+			bor.setName(rs.getString("name"));
+			bor.setAddress(rs.getString("address"));
+			bor.setPhone(rs.getString("phone"));
+			
+		}
+		return bor;
 	}
+	
+	
+	
+	//show branch for checkout 
+	public List<LibraryBranch>  displayBranchCheckout() throws SQLException {
+		String sql = "SELECT * from tbl_library_branch";
+		PreparedStatement ps = getConnection().prepareStatement(sql);
+		ResultSet rs = ps.executeQuery();
+
+		 List<LibraryBranch> bch = new ArrayList<LibraryBranch>();
+		 
+		 while (rs.next()) {
+			 LibraryBranch branch = new LibraryBranch();
+			 branch.setBranchId(rs.getInt("branchId"));
+			 branch.setBranchName(rs.getString("branchName"));
+			 branch.setBranchAddress(rs.getString("branchAddress"));
+			 bch.add(branch);	
+			}
+			return bch;
+		}
+	
+	
+	
+	//show  book for checkout
+	public List<Book>  displayBookCheckout(int branchId) throws SQLException {
+		String sql= ("SELECT tbl_book.bookId, CONCAT(tbl_book.title, ' by ' , tbl_author.authorName) AS title  FROM tbl_book INNER JOIN tbl_author ON tbl_book.authId = tbl_author.authorId INNER JOIN tbl_book_copies ON tbl_book.bookId =  tbl_book_copies.bookId INNER JOIN tbl_library_branch ON tbl_book_copies.branchId = tbl_library_branch.branchId WHERE tbl_library_branch.branchId = ? AND tbl_book_copies.noOfCopies >=1 ");
+		
+		PreparedStatement ps = getConnection().prepareStatement(sql);
+		ps.setInt (1, branchId);
+		ResultSet rs = ps.executeQuery();
+
+		 List<Book> bk = new ArrayList<Book>();
+		 
+		 while (rs.next()) {
+			 Book book = new Book();
+			 book.setBookId(rs.getInt("bookId"));
+			 book.setTitle(rs.getString("title"));
+			 bk.add(book);	
+			}
+			return bk;
+		}
+	
+	
+	
+    
+    //show book to return
+    
+    public List<Book>  displayBookReturn(int branchId, int cardNo) throws SQLException {
```

Comment:
Catch your exceptions here in the DAO and you will not have to throw them in your service and controller classes.